oauthproxy.go

```diff
@@ -108,17 +109,14 @@ type OAuthProxy struct {
 	PassAccessToken         bool
 	SetAuthorization        bool
 	PassAuthorization       bool
+	SkipAuthStripHeaders    bool
 	PreferEmailToUser       bool
-	skipAuthRegex           []string
-	skipAuthPreflight       bool
-	skipAuthStripHeaders    bool
 	skipJwtBearerTokens     bool
 	mainJwtBearerVerifier   *oidc.IDTokenVerifier
 	extraJwtBearerVerifiers []*oidc.IDTokenVerifier
-	compiledRegex           []*regexp.Regexp
 	templates               *template.Template
 	realClientIPParser      ipapi.RealClientIPParser
-	trustedIPs              *ip.NetSet
+	rulesEngine             *engine.RulesEngine
```

Comment:
Maybe call this something more related to what it is used for? `requestAuthorization`?